src/mmw/apps/bigcz/clients/cinergi.py

```diff
@@ -5,40 +5,88 @@
 
 import requests
 import dateutil.parser
+from django.contrib.gis.geos import Polygon
 
 from apps.bigcz.models import Resource, ResourceLink, ResourceList, BBox
 
 
 CATALOG_NAME = 'cinergi'
-GEOPORTAL_URL = 'http://cinergi.sdsc.edu/geoportal/rest/find/document'
+GEOPORTAL_URL = 'http://132.249.238.169:8080/geoportal/opensearch'
 
 
-def parse_links(item):
-    result = []
-    links = item['links']
-    if links:
-        for link in links:
-            result.append(ResourceLink(link['type'], link['href']))
-    return result
+def parse_date(value):
+    return dateutil.parser.parse(value)
 
 
-def parse_date(value):
-    try:
-        # Some dates are 0-based?
-        return dateutil.parser.parse(value)
-    except ValueError:
+def parse_envelope(envelope):
+    if 'coordinates' not in envelope:
+        raise ValueError('Expected envelope to contain coordinates')
+
+    coords = envelope['coordinates']
+
+    northwest = coords[0]
+    southeast = coords[1]
+
+    xmin, ymin = northwest
+    xmax, ymax = southeast
+
+    return Polygon.from_bbox((xmin, ymin, xmax, ymax))
+
+
+def parse_bbox(source):
+    """
+    Parse bbox from item source by doing a unary union of each
+    available envelope_geo object.
+
+    - envelope_geo may be absent, an object, or list of objects
+    """
+    envelope = source.get('envelope_geo')
+
+    if not envelope:
         return None
 
+    if isinstance(envelope, dict):
+        envelope = [envelope]
+
+    poly = parse_envelope(envelope[0])
+
+    for item in envelope[1:]:
+        poly |= parse_envelope(item)
+
+    # xmin, ymin, xmax, ymax
+    return poly.extent
```

Comment:
This implementation will change after #1940. Instead of returning a single bounding box per result, we should return `Point`, `Polygon`, or `MultiPolygon` GeoJSON.